src/explorer.ts

```diff
@@ -119,21 +124,29 @@ class KubernetesContext implements KubernetesObject {
 
     getTreeItem(): vscode.TreeItem | Thenable<vscode.TreeItem> {
         const treeItem = new vscode.TreeItem(this.id, vscode.TreeItemCollapsibleState.Collapsed);
-        treeItem.contextValue = "vsKubernetes.cluster";
-        treeItem.iconPath = vscode.Uri.file(path.join(__dirname, "../../images/k8s-logo.png"));
+        treeItem.contextValue = this.clusterType;
+
+        let iconPath: vscode.Uri;
```

Comment:
Consider breaking this block out into a separate function that returns the icon path, so we can make this `const` - a la `const iconPath = getIconPathForClusterType();`

Reply:
fixed.